### PR TITLE
Modularize and Modernize (ES6)

### LINKED ISSUE #25: Modularize
Please complete [#24](https://github.com/ericelliott/cuid/issues/24) first.

## Stop using the hacky concat build.

* [x] Use ES6 modules
* [x] Drop applitude support (it's obsolete)
* [x] Use separate client/server build entry points and share common features between them.

```
.
└── cuid
    ├── build
    │   ├── client
    │   └── server
    ├── source
    │   ├── client
    │   └── server
    └── test
        ├── client
        └── server
```

## Ignore build directory in diffs

You can [ignore build files in diffs](https://github.com/gitlabhq/gitlabhq/issues/6916#issuecomment-62520577).

Create a .gitattributes file in the repo root:

```
build    -diff
```

### PR DESCRIPTION
**Pull request by therealklanni**
Premature PR so you can give early feedback

Closes #25

### CODE REVIEW

**Comment by therealklanni** [comment]
Holding off on pushing the modularized code until I hear about https://github.com/ericelliott/cuid/issues/25#issuecomment-118561008

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (ericelliott) [inline .eslintrc]: Please use the file from [prod-module-boilerplate](https://github.com/ericelliott/prod-module-boilerplate/blob/master/.eslintrc)

File: grunt.js [hunk 2]
```diff
@@ -1,52 +0,0 @@
-/*global module*/
```
Comment (ericelliott) [inline grunt.js]: Hurray! Grunt is dead!

File: lib/applitude.js [hunk 3]
```diff
@@ -1,938 +0,0 @@
-;!function(exports, undefined) {
```
Comment (ericelliott) [inline lib/applitude.js]: Hurray, applitude is dead!

File: package.json [hunk 4]
```diff
@@ -27,10 +27,12 @@
     "test": "./test"
   },
   "devDependencies": {
```
Comment (ericelliott) [inline package.json]: :guitar: :fire:

File: src/footer.js [hunk 5]
```diff
@@ -1,11 +0,0 @@
-
```
Comment (ericelliott) [inline src/footer.js]: I'm so excited to see this ugly hack go. :+1:

**Comment by ericelliott** [comment]
Don't you just love deleting things? =)

Now we just need the new modular hotness!

**Comment by therealklanni** [comment]
Yep! Almost there, just need to switch from tape to blue-tape. Any objection to writing the saucelabs integration yourself? I assume you have an account already, so it would probably make more sense for you to set it up with your account.

File: .eslintrc (hunk 1 above)
Comment (therealklanni) [inline .eslintrc]: No problemo

File: package.json (hunk 4 above)
Comment (therealklanni) [inline package.json]: :metal:

**Comment by ericelliott** [comment]
[Sauce Labs is free for Open Source](https://saucelabs.com/opensource)

File: source/index.js [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+ *
+ * Extracted from CLCTR
+ *
+ * Copyright (c) Eric Elliott 2012
+ * MIT License
+ */
+
+'use strict';
+
+let c = 0;
```
Comment (ericelliott) [inline source/index.js]: Use const for all these... don't use the UGLY_UPPERCASE_CONVENTION. I do everything const by default, and I don't like source files yelling at me. ;)

File: source/server/index.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+import createCuid from '../index.js';
+const fingerprint = require('node-fingerprint')();
+
+let { cuid, slug } = createCuid(fingerprint);
```
Comment (ericelliott) [inline source/server/index.js]: const

File: source/client/index.js [hunk 8]
```diff
@@ -0,0 +1,7 @@
+'use strict';
+
+import createCuid from '../index.js';
+const fingerprint = require('browser-fingerprint')();
+
+let { cuid, slug } = createCuid(fingerprint);
```
Comment (ericelliott) [inline source/client/index.js]: const

File: test/server/index.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+import { cuid, slug } from '../../source/server/index.js';
+import test from 'blue-tape';
+
+let MAX = 1200000;
```
Comment (ericelliott) [inline test/server/index.js]: const

File: test/server/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+let collisionTest = (fn) => {
```
Comment (ericelliott) [inline test/server/index.js]: const

**Comment by therealklanni** [comment]
OK, I didn't see anything saying it was free when I looked. Thanks for the link.

**Comment by ericelliott** [comment]
:+1:

**Comment by therealklanni** [comment]
> If that's too much trouble, let me know, I'll open a separate issue for it. =)

I'll take a crack at it :)

**Comment by ericelliott** [comment]
Let's go ahead and merge this and finish up Sauce Labs in another PR. =)

**Comment by therealklanni** [comment]
Sounds good. I was just getting it started, but now I want to take a break from it :sleepy:

**Comment by ericelliott** [comment]
Yeah, enjoy some of the weekend. =)

**Comment by therealklanni** [comment]
You'd be surprised at how much I got done today that *wasn't* programming :+1:

**Comment by ericelliott** [comment]
:cake:

**Pull request merged**